app_dart/lib/src/request_handlers/check_for_waiting_pull_requests.dart

```diff
@@ -162,50 +162,51 @@ class CheckForWaitingPullRequests extends ApiRequestHandler<Body> {
     }
-    final String labelId = label['id'] as String;
+    final String labelId = data.repository.labels.nodes[0].id;
     log.info('LabelId of returned PRs: $labelId');
     final List<_AutoMergeQueryResult> list = <_AutoMergeQueryResult>[];
-    final Iterable<Map<String, dynamic>> pullRequests =
-        (label['pullRequests']['nodes'] as List<dynamic>).cast<Map<String, dynamic>>();
-    for (Map<String, dynamic> pullRequest in pullRequests) {
-      final Map<String, dynamic> commit = pullRequest['commits']['nodes'].single['commit'] as Map<String, dynamic>;
+    final List<$LabeledPullRequestsWithReviews$repository$labels$nodes$pullRequests$nodes> pullRequests =
+        data.repository.labels.nodes[0].pullRequests.nodes;
+    for ($LabeledPullRequestsWithReviews$repository$labels$nodes$pullRequests$nodes pullRequest in pullRequests) {
+      final $LabeledPullRequestsWithReviews$repository$labels$nodes$pullRequests$nodes$commits$nodes$commit commit =
+          pullRequest.commits.nodes[0].commit;
       // Skip commits that are less than an hour old.
       // Use the committedDate if pushedDate is null (commitedDate cannot be null).
-      final DateTime utcDate =
-          DateTime.parse(commit['pushedDate'] as String ?? commit['committedDate'] as String).toUtc();
+      final DateTime utcDate = DateTime.parse((commit.pushedDate ?? commit.committedDate).value).toUtc();
       if (utcDate.add(const Duration(hours: 1)).isAfter(DateTime.now().toUtc())) {
         continue;
       }
-      final String author = pullRequest['author']['login'] as String;
-      final String id = pullRequest['id'] as String;
-      final int number = pullRequest['number'] as int;
+      final String author = pullRequest.author.login;
+      final String id = pullRequest.id;
+      final int number = pullRequest.number;
 
       final Set<String> changeRequestAuthors = <String>{};
       final bool hasApproval = config.rollerAccounts.contains(author) ||
           _checkApproval(
-            (pullRequest['reviews']['nodes'] as List<dynamic>).cast<Map<String, dynamic>>(),
+            pullRequest.reviews.nodes,
             changeRequestAuthors,
           );
 
-      final String sha = commit['oid'] as String;
-      final List<Map<String, dynamic>> statuses =
-          (commit['status']['contexts'] as List<dynamic>).cast<Map<String, dynamic>>();
-      List<Map<String, dynamic>> checkRuns;
-      if (commit['checkSuites']['nodes'] != null && (commit['checkSuites']['nodes'] as List<dynamic>).isNotEmpty) {
-        checkRuns =
-            (commit['checkSuites']['nodes']?.first['checkRuns']['nodes'] as List<dynamic>).cast<Map<String, dynamic>>();
-      }
-      checkRuns = checkRuns ?? <Map<String, dynamic>>[];
+      final String sha = commit.oid.value;
+      final List<
+              $LabeledPullRequestsWithReviews$repository$labels$nodes$pullRequests$nodes$commits$nodes$commit$status$contexts>
```

Comment:
These seem unfortunate. I wonder if there's any setting to have it generate better names than this?

Reply:
I asked in the gql discord chat room and the answer seems to be know. For this issue, our best hope is for https://github.com/dart-lang/language/issues/65 to be fixed.

For now, I can also use `// ignore: always_specify_types` to save as many type names as possible. Please see my latest patch.